services/auth.service.js

```diff
@@ -1,17 +1,43 @@
 import { EntityService } from './entity.service.js';
-
+import bcrypt from 'bcrypt';
 export class AuthService {
   constructor(db) {
     this.service = new EntityService(db, 'user');
   }
 
   async logIn(email, password) {
-    const user = await this.service.getOne({ email, password });
+    const user = await this.service.getOne({ email });
 
     if (!user) {
       throw new Error('Wrong credentials');
     }
 
+    const checkPassword = await bcrypt.compare(password, user.password);
+
+    if (!checkPassword) {
+      throw new Error(`Wrong credentials`);
+    }
+
     return user;
   }
+
+  async register(email, password, username) {
+    const user = await this.service.getOneConditional({ email, username });
+
+    if (user) {
```

Comment:
you can use ? operator avoid getting an error "cannot read properties of undefined".
Something like:
```js
if (user?.username === username) {
     throw new Error(`Username "${username}" is already taken`);
}
```